pkg/kv/kvserver/replica_application_state_machine.go

```diff
@@ -259,12 +216,12 @@ func (b *replicaAppBatch) Stage(
 ) (apply.CheckedCommand, error) {
 	cmd := cmdI.(*replicatedCmd)
 	if cmd.Index() == 0 {
-		return nil, makeNonDeterministicFailure("processRaftCommand requires a non-zero index")
+		return nil, kvserverbase.NewNonDeterministicFailureErrorf("processRaftCommand requires a non-zero index")
 	}
 	if idx, applied := cmd.Index(), b.state.RaftAppliedIndex; idx != applied+1 {
 		// If we have an out of order index, there's corruption. No sense in
 		// trying to update anything or running the command. Simply return.
-		return nil, makeNonDeterministicFailure("applied index jumped from %d to %d", applied, idx)
+		return nil, kvserverbase.NewNonDeterministicFailureErrorf("applied index jumped from %d to %d", applied, idx)
```

Comment:
nit: remove New prefix? For consistency with the `Wrapf` version of this func, and for brevity.

Reply:
Good idea, will do. Btw, this would have been a good push.